pkg/util/functions/functions.go

```diff
@@ -0,0 +1,38 @@
+package functions
+
+// RunSequentially executes a series of functions sequentially. Each function returns a boolean
+// indicating if the function was successfully, and an error indicating if something went wrong.
+// if any function returns an error, an early exit happens. The first parameter indicates if the functions
+// should run in the order provided. A value of true indicates they should run in reverse.
+func RunSequentially(reversed bool, funcs ...func() (bool, error)) (bool, error) {
```

Comment:
[q] why are we using a boolean that tells us if we want to flip the order of execution?

I'm thinking of how a dev would interpret seeing this function called:

`RunSequentially(true, functions...)`

would make me thing that these will be ran in order, not in the opposite one.

Reply:
ah, yes that probably makes more sense! I'll do that refactor 👍